cpp/src/arrow/type.h

```diff
@@ -1223,6 +1223,35 @@ class ARROW_EXPORT DenseUnionType : public UnionType {
   std::string name() const override { return "dense_union"; }
 };
 
+/// \brief Type class for run-end encoded data
+class ARROW_EXPORT RunEndEncodedType : public NestedType {
+ public:
+  static constexpr Type::type type_id = Type::RUN_END_ENCODED;
+
+  static constexpr const char* type_name() { return "run_end_encoded"; }
+
+  explicit RunEndEncodedType(std::shared_ptr<DataType> run_end_type,
+                             std::shared_ptr<DataType> value_type);
+
+  DataTypeLayout layout() const override {
+    // NOTE(felipecrv): his can be removed once we ensure existing code does not
+    // assume the existence of at least one buffer.
```

Comment:
This seems like a huge change that would break compatibility in many places. That's why I opted for a `NOTE`. I could reword it as a comment without hinting that this will actually change someday.

Reply:
The current wording sounds like you intend to change it.

Reply:
Changed it to a comment.